### PR TITLE
Ignore hidden files & folders

### PR DESCRIPTION
**Pull request by AyhamAl-Ali**
**Redo old PR #3911 due to the recent changes in ScriptLoader (Section API)**

### Description

This only makes Skript ignore folders that begins with a dot when:
- Loading scripts folder at startup.
- Reloading a parent folder that doesn't begin with a dot but has a child folder that begins with a dot.
- Reloading all scripts.

But doesn't work when:
- Reloading specific file inside a folder that begins with a dot (that means the script file will load)
- Reloading specific folder that begins with a dot (that means all the scripts inside that folder will load except folders inside it that begins with a dot)

Therefor, SkriptCommand.class needs a rework to prevent reloading folders that begins with a dot or files that any of their parents begin with a dot and support them with enable and disable args.

**Also**, file/folder permissions should also be worked on but since it is not a high priority due to the low chances of it happening (and probably caused by the user side) but I have found that `Files#walk` causes this issue due to it reading attributes of the files regardless of their permissions, I have tried to fix it but seems that it isn't as easy as I thought due to `walk` method not having an easy way to check for perms before reading it.

---
**Target Minecraft Versions:** 
**Requirements:** 
**Related Issues:** #3910 #3866

### CODE REVIEW

**Review by APickledWalrus — APPROVED** [review]
This looks fine to me - sorry you had to go so long without a review! I don't know if we necessarily have to handle the issue with Files#walk right now - it could always be done in a later PR as it's unlikely to occur like you said.

**Comment by AyhamAl-Ali** [comment]
> This looks fine to me - sorry you had to go so long without a review! I don't know if we necessarily have to handle the issue with Files#walk right now - it could always be done in a later PR as it's unlikely to occur like you said.

Thank you for the review. Regarding the Files#walk yes it can be handled later in another PR, as for now this is only for the hidden files/folders.

File: src/main/java/ch/njol/skript/ScriptLoader.java [hunk 1]
```diff
@@ -217,21 +217,21 @@ private static ParserInstance getParser() {
 	 */
 	private static final FileFilter scriptFilter =
 		f -> f != null
-			&& (f.isDirectory() || StringUtils.endsWithIgnoreCase(f.getName(), ".sk"))
-			&& !f.getName().startsWith("-");
-	
+			&& (f.isDirectory() && !f.getName().startsWith(".") || !f.isDirectory() && StringUtils.endsWithIgnoreCase("" + f.getName(), ".sk"))
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/ScriptLoader.java:220]: Why the `"" + ` here?

Comment (AyhamAl-Ali) [reply src/main/java/ch/njol/skript/ScriptLoader.java]: It was probably left over that I forgot about, should be good now :+1:

**Pull request merged**